distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso

```diff
@@ -66,11 +101,45 @@ type Bytes
             ".dat" -> Bytes
             _ -> Nothing
 
+    ## If the File_Format supports reading from the web response, return a configured instance.
+    for_web : Text -> URI -> Bytes | Nothing
+    for_web _ _ = Nothing
+
     ## PRIVATE
        Implements the `File.read` for this `File_Format`
     read : File -> Problem_Behavior -> Any
     read self file _ =
         file.read_bytes
 
+type JSON_File
+    ## If the File_Format supports reading from the file, return a configured instance.
+    for_file : File -> JSON_File | Nothing
+    for_file file =
+        case file.extension of
+            ".json" -> JSON_File
+            ".geojson" -> JSON_File
+            _ -> Nothing
+
+    ## If the File_Format supports reading from the web response, return a configured instance.
+    for_web : Text -> URI -> JSON_File | Nothing
+    for_web content_type _ =
+        first = content_type.split ';' . first
+        case first of
+            "application/json" -> JSON_File
+            _ -> Nothing
```

Comment:
For `plain/text` we also had `.trim`, here we do not. Why? We probably should be consistent.